application/src/main/java/org/togetherjava/tjbot/features/roleapplication/ApplicationApplyHandler.java

```
@@ -0,0 +1,142 @@
+    private final Predicate<String> applicationChannelPattern;
+    private final RoleApplicationSystemConfig roleApplicationSystemConfig;
+
+    /**
+     * Constructs a new {@code ApplicationApplyHandler} instance.
+     *
+     * @param roleApplicationSystemConfig the configuration that contains the details for the
+     *        application form including the cooldown duration and channel pattern.
+     */
+    public ApplicationApplyHandler(RoleApplicationSystemConfig roleApplicationSystemConfig) {
+        this.roleApplicationSystemConfig = roleApplicationSystemConfig;
+        this.applicationChannelPattern =
+                Pattern.compile(roleApplicationSystemConfig.submissionsChannelPattern())
+                    .asMatchPredicate();
+
+        final Duration applicationSubmitCooldownDuration =
+                Duration.ofMinutes(roleApplicationSystemConfig.applicationSubmitCooldownMinutes());
+        applicationSubmitCooldown =
+                Caffeine.newBuilder().expireAfterWrite(applicationSubmitCooldownDuration).build();
+    }
+
+    /**
+     * Sends the result of an application submission to the designated application channel in the
+     * guild.
+     * <p>
+     * The {@code args} parameter should contain the applicant's name and the role they are applying
+     * for.
+     *
+     * @param event the modal interaction event triggering the application submission
+     * @param args the arguments provided in the application submission
+     * @param answer the answer provided by the applicant to the default question
+     */
+    protected void sendApplicationResult(final ModalInteractionEvent event, List<String> args,
+            String answer) {
+        Guild guild = event.getGuild();
+        if (args.size() != 2 || guild == null) {
+            return;
+        }
+
+        Optional<TextChannel> applicationChannel = getApplicationChannel(guild);
+        if (applicationChannel.isEmpty()) {
+            return;
+        }
```

warning instead of silent failure?